files/www/cgi-bin/status

```diff
@@ -51,6 +51,42 @@ $wifi_iface = get_interface("wifi");
 $wifi_iface =~ /wlan(\d+)/;
 $radio = ( defined $1 )? "radio$1" : "radio0";
 $wifi_disable = ( $wifi_iface =~ /eth.*$/ )? 1 : 0;
+$fw = `cat /etc/mesh-release`;
+$nightly = `/sbin/uci -q get aredn.\@downloads[0].nightly`;
+
+# Check to see if the node is on a nightly build, and if not proceed in loop
+if($nightly != 1) {
+
+	# Check if there is an update to the firmware
+	if(-e '/tmp/web/firmware.list') {
+
+		# See if new file should be downloaded, and if so, set a variable to do this at the end of the script
```

Comment:
i think this logic would be better in a separate script that is called by a cron job instead of everytime the status page is loaded.

Reply:
(sorry, wrong code snipped selected.  was referencing the date checking code.)

Reply:
My thinking was that the status page, even on active nodes, probably isn't loaded all that often, and that info is really only valuable when the page is being loaded.  A cron job would have it run all the time in the background, whereas this will only check when there is a user that might actually view and/or do something with the info.

As a more concrete example, if a node is put up and no one ever logs into it but it is very busy passing traffic away in the background, the way I wrote this it would have no impact whatsoever on the node nor the aredn website.  However, if a user logs in and starts navigating around, it will check once and only once, if there is an update, and prompt the user accordingly.  If they then don't log in again for another year, again no impact for another year.

It isn't a ton of traffic / bits, but every time it pings home, it is downloading that file from the AREDN servers as well.  This way it would, at most, download the file once a week, and at least it would never download the file at all.  And when the download does happen, it would be at a totally random time.  If we did a cron job, you might see 1000 hits to that file every Monday morning at 4am west coast time for example, even if no user will see that info before the next update on any given node.

If a cron job is preferable by the team that would be a relatively easy change, however the logic I noted above was why I believe the current setup (or something like it) is preferable to a cron job.

Reply:
agreed.  good point.